zipline/utils/calendars/exchange_calendar.py

```diff
@@ -0,0 +1,503 @@
+#
+# Copyright 2016 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import (
+    ABCMeta,
+    abstractproperty,
+    abstractmethod,
+)
+
+import pandas as pd
+from pandas import (
+    DataFrame,
+    date_range,
+    DateOffset,
+    DatetimeIndex,
+)
+from pandas.tseries.offsets import CustomBusinessDay
+from six import with_metaclass
+
+from zipline.errors import (
+    InvalidCalendarName,
+    CalendarNameCollision,
+)
+from zipline.utils.memoize import remember_last
+
+from .calendar_helpers import (
+    next_scheduled_day,
+    previous_scheduled_day,
+    next_open_and_close,
+    previous_open_and_close,
+    scheduled_day_distance,
+    minutes_for_day,
+    days_in_range,
+    minutes_for_days_in_range,
+    add_scheduled_days,
+    all_scheduled_minutes,
+    next_scheduled_minute,
+    previous_scheduled_minute,
+    minute_window,
+)
+
+start_default = pd.Timestamp('1990-01-01', tz='UTC')
+end_base = pd.Timestamp('today', tz='UTC')
+# Give an aggressive buffer for logic that needs to use the next trading
+# day or minute.
+end_default = end_base + pd.Timedelta(days=365)
```

Comment:
That being said, if this is a sufficiently special case, it probably makes sense to keep the default at a 1 year buffer and specify the end date in this particular instance.